src/store/annotation.ts

```diff
@@ -558,6 +558,24 @@ export class Annotations extends VuexModule {
     this.setSelected([]);
   }
 
+  @Action
+  public async deleteUnselectedAnnotations() {
+    const unselectedIds = this.annotations
+      .filter(
+        (annotation) =>
+          !this.selectedAnnotations.some(
+            (selected) => selected.id === annotation.id,
+          ),
+      )
+      .map((annotation) => annotation.id);
+
+    if (unselectedIds.length === 0) {
+      return;
+    }
```

Comment:
I would put this at the beginning of `deleteAnnotations`  
I think it would be a better responsibility share

Reply:
Good point!